modin/pandas/utils.py

```diff
@@ -345,6 +340,28 @@ def broadcast_item(
     )
     to_shape = new_row_len, new_col_len
 
+    item = _maybe_reindex_item(obj, row_lookup, col_lookup, item, need_columns_reindex)
+
+    try:
+        item = np.array(item)
+        if np.prod(to_shape) == np.prod(item.shape):
+            return item.reshape(to_shape)
+        else:
+            return np.broadcast_to(item, to_shape)
+    except ValueError:
+        from_shape = np.array(item).shape
+        raise ValueError(
+            f"could not broadcast input array from shape {from_shape} into shape "
+            + f"{to_shape}"
+        )
+
+
+def _maybe_reindex_item(obj, row_lookup, col_lookup, item, need_columns_reindex=True):
```

Comment:
Can we remove the "maybe" and make this method only valid for series and dataframes?